lib/server.js

```diff
@@ -253,24 +262,26 @@ module.exports = function createServer(config, log) {
               blockReason = blockReasons.IP_BAD_REPUTATION
             }
 
+            const result = {
+              block: block,
+              blockReason: blockReason,
+              retryAfter: retryAfter,
+              unblock: canUnblock,
+              suspect: suspect
+            }
+
+            if (allowWhitelisted(result, ip, email)) {
+              // Prevent white-listed email addresses from blocking
+              // subsequent requests to /checkIpOnly
+              ipRecord.clear()
+            }
+
             return setRecords(ip, ipRecord, email, emailRecord, ipEmailRecord, phoneNumber, smsRecord)
-              .then(
-                function () {
-                  return {
-                    block: block,
-                    blockReason: blockReason,
-                    retryAfter: retryAfter,
-                    unblock: canUnblock,
-                    suspect: suspect
-                  }
-                }
-              )
+              .then(() => result)
           }
         )
         .then(
           function (result) {
-            allowWhitelisted(result, ip, email)
```

Comment:
This had to move to before `setRecords` because we want to write the cleared ip record to the cache.